### PR TITLE
Binary Search approach

### PR DESCRIPTION
**Pull request by bobahop**
Approach for Binary Search

### CODE REVIEW

File: exercises/practice/binary-search/.approaches/while-with-if-statements/content.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+```
+
+This approach starts by checking for a `NULL` input array.
+It then defines the [`size_t`][size_t] variables to keep track of the `left` and `right` positions being searched for in the array.
+The `left` is initialized to `0` and the `right` is initialized to the length passed in.
+
+The while loop will loop as long as the `left` position is less than the `right` position.
+
+Inside the loop, the middle value is set by `left` plus ((`right` - `left`) divided by `2`).
+- For example, if `left` is `0` and `right` is `10`, then the middle is calculated to `5`.
```
Comment (wolf99) [inline exercises/practice/binary-search/.approaches/while-with-if-statements/content.md:42]: Need blank line before list

**Comment by wolf99** [comment]
Just the one nit, otherwise it looks great! 👍

**Comment by bobahop** [comment]
@wolf99: Thanks, the suggested change was committed.

**Comment by wolf99** [comment]
Sorry bob, the comments have been just comments, not changes.
I'm on mobile app so just jotting comments as I see things.

I think you've been marking comments as resolved rather than committing changes.

Either way it looks like there's no change since I reviewed last.

File: exercises/practice/binary-search/.approaches/while-with-if-statements/content.md (hunk 1 above)
Comment (bobahop) [reply exercises/practice/binary-search/.approaches/while-with-if-statements/content.md:43]: Whoa, sorry! Not used to reviewers doing that. Okay, change made.

Luckily, I think other changes I may have missed on previous exercises were fairly minor formatting issues.

**Comment by bobahop** [comment]
@wolf99: Made the change here and made a couple PRs for the changes I missed in previous approaches.

**Review by wolf99 — APPROVED** [review]

**Comment by wolf99** [comment]
All merged, thanks!

**Pull request merged**